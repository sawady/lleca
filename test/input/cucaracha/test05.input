

fun f() {
  putChar(104)
  putChar(111)
  putChar(108)
  putChar(97)
  putChar(10)
  g()
}

fun g() {
  putChar(99)
  putChar(104)
  putChar(97)
  putChar(117)
  putChar(10)
}

fun main() {
  f()
  f()
  g()
}
