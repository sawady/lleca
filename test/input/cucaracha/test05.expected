Cons(
  Function(
    f,
    Nil,
    Unit,
    Cons(
      StmtCall(
        putChar,
        Cons(
          ExprConstNum(
            104
          ),
          Nil
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              111
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                108
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  97
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    10
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  g,
                  Nil
                ),
                Nil
              )
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      g,
      Nil,
      Unit,
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              99
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                104
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  97
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    117
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                ),
                Nil
              )
            )
          )
        )
      )
    ),
    Cons(
      Function(
        main,
        Nil,
        Unit,
        Cons(
          StmtCall(
            f,
            Nil
          ),
          Cons(
            StmtCall(
              f,
              Nil
            ),
            Cons(
              StmtCall(
                g,
                Nil
              ),
              Nil
            )
          )
        )
      ),
      Nil
    )
  )
)
